src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/filter_table.go

```diff
@@ -62,14 +71,31 @@ func (s *filterStatements) prepare(db *sql.DB) (err error) {
 
 func (s *filterStatements) selectFilter(
 	ctx context.Context, localpart string, filterID string,
-) (filter string, err error) {
+) (filter []byte, err error) {
 	err = s.selectFilterStmt.QueryRowContext(ctx, localpart, filterID).Scan(&filter)
 	return
 }
 
 func (s *filterStatements) insertFilter(
-	ctx context.Context, filter string, localpart string,
-) (pos string, err error) {
-	err = s.insertFilterStmt.QueryRowContext(ctx, filter, localpart).Scan(&pos)
+	ctx context.Context, filter []byte, localpart string,
+) (pos []byte, err error) {
+	var existingFilter []byte
```

Comment:
Hang on, are we returning the filter ID or the actual filter? Might be worth documenting the return type of this function

Reply:
Ah, your correct, we should be returning the ID from this method, not the filter itself. Good catch!